plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfigurationProvider.java

```diff
@@ -109,6 +110,11 @@ public void updateConfiguration(Configuration configuration, HdfsContext context
             hasher.putString(role, UTF_8);
         });
 
+        mapping.getKmsKeyId().ifPresent(key -> {
+            configuration.set(S3_KMS_KEY_ID, key);
+            hasher.putString(key, UTF_8);
```

Comment:
add "kms key id" to hash as well (as per previous GK's comment)

Reply:
Oh - good catch - forgot about that. Sorry

Reply:
Technically the comment also applies for other entries put in hasher. Though crafting a collision should not be something observable in the real world.